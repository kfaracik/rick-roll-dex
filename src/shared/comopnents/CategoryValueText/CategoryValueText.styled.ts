import {StyleSheet} from 'react-native';
import {Colors} from '../../utils/colors';

export const styles = StyleSheet.create({
  container: {
    padding: 8,
  },
  categoryNameText: {
    color: Colors.mediumGreen,
    fontSize: 12,
  },
  valueText: {
    color: Colors.darkGreen,
    fontSize: 16,
  },
});
