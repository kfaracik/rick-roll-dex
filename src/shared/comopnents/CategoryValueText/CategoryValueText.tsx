import {
  View,
  Text,
  type ViewStyle,
  type TextStyle,
  type StyleProp,
} from 'react-native';
import React from 'react';
import {styles} from './CategoryValueText.styled';

type CategoryValueTextProps = {
  category: string;
  value: string;
  style?: StyleProp<ViewStyle>;
  valueTextStyle?: StyleProp<TextStyle>;
};

export const CategoryValueText = ({
  category,
  value,
  style,
  valueTextStyle,
}: CategoryValueTextProps) => {
  return (
    <View style={[styles.container, style]}>
      <Text style={styles.categoryNameText}>{category}</Text>
      <Text style={[styles.valueText, valueTextStyle]}>{value}</Text>
    </View>
  );
};
