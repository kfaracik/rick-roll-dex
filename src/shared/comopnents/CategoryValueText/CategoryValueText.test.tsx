import React from 'react';
import {render} from '@testing-library/react-native';
import {CategoryValueText} from './CategoryValueText';

describe('CategoryValueText component', () => {
  it('should generate snapshot', () => {
    const {toJSON} = render(
      <CategoryValueText category={'Category'} value={'some calue 1234'} />,
    );
    expect(toJSON()).toMatchSnapshot();
  });
});
