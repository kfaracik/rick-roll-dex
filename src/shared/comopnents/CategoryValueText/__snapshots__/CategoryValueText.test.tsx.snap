// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should generate snapshot 1`] = `
<View
  style={
    [
      {
        "padding": 8,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      {
        "color": "#59695C",
        "fontSize": 12,
      }
    }
  >
    Category
  </Text>
  <Text
    style={
      [
        {
          "color": "#162C1B",
          "fontSize": 16,
        },
        undefined,
      ]
    }
  >
    some calue 1234
  </Text>
</View>
`;

exports[`CategoryValueText component should generate snapshot 1`] = `
<View
  style={
    [
      {
        "padding": 8,
      },
      undefined,
    ]
  }
>
  <Text
    style={
      {
        "color": "#59695C",
        "fontSize": 12,
      }
    }
  >
    Category
  </Text>
  <Text
    style={
      [
        {
          "color": "#162C1B",
          "fontSize": 16,
        },
        undefined,
      ]
    }
  >
    some calue 1234
  </Text>
</View>
`;
